src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/StatusCheckFunction.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.DurableTask.Client;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+
+public class StatusCheckFunction
+{
+    private static readonly OrchestrationQuery ActiveOrchestrationsQuery = new()
+    {
+        Statuses = new List<OrchestrationRuntimeStatus>
+        {
+            OrchestrationRuntimeStatus.Pending,
+            OrchestrationRuntimeStatus.Running
+        }
+    };
+
+    [Function("StatusCheck")]
+    [Produces("application/json")]
+    public static async Task<IActionResult> StatusCheck(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get")]
+#pragma warning disable IDE0060
+        HttpRequest request,
+#pragma warning restore IDE0060
+        [DurableClient] DurableTaskClient client)
+    {
+        var activeOrchestrations = await client
+            .GetAllInstancesAsync(filter: ActiveOrchestrationsQuery)
+            .ToListAsync();
+        return new OkObjectResult(new { ActiveOrchestrations = activeOrchestrations.Count });
```

Comment:
Would it be better to return a 5xx status code if there are active orchestrations?

The main benefit is that we could reuse the `wait-for-endpoint-success.yml` task rather than need the additional `wait-for-orchestrations-to-complete.yml` task (which basically does a similar thing).

We'd probably want to also rename the function to something more intuitive for these semantics e.g. `DeployStatusCheck` (checks if the function is ready for a new deployment).

Reply:
I see what you're saying, but I'm slightly hesitant to do this because I feel that the details you get form this endpoint could be useful, and also there's no easy way to distinguish a planned 500 from an ISE 500 if one should arise for whatever reason.  

The other reason why having the count and the specific pipeline job that's asserting on the count is useful I think is that you can see during deploy in the DevOps logging how many active orchestrations are currently holding up the deploy, which gives you some indication of how long it'll take to get through.

It's not necessarily so desperately urgent at the mo because we're expecting orchestrations to be relatively quick to run, but if we end up reusing for another durable function app or we include longer orchestrations here in the future, I think it would be beneficial to have that visibility.

What do you reckon?

Reply:
(see the screenshot under `Testing pipeline changes with long-running orchestration` in the PR description for an idea of what I mean)

Reply:
Hmm, there's potentially some alterations we could do to make this work i.e. we could change `wait-for-endpoint-success.yml` to output the body of non-200 responses. This would output the number of active orchestrations in the response JSON.

I'd also probably lean towards using 503 instead of 500 as this might be better for communicating the service isn't available to be deployed.

Regardless, it's not a big deal either way. If you're happy enough with the current approach, happy for this to go in as-is!